homeassistant/components/ccm15/coordinator.py

```diff
@@ -0,0 +1,86 @@
+"""Climate device for CCM15 coordinator."""
+import datetime
+import logging
+
+from ccm15 import CCM15Device, CCM15DeviceState, CCM15SlaveDevice
+import httpx
+
+from homeassistant.components.climate import HVACMode
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONST_FAN_CMD_MAP,
+    CONST_STATE_CMD_MAP,
+    DEFAULT_INTERVAL,
+    DEFAULT_TIMEOUT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class CCM15Coordinator(DataUpdateCoordinator[CCM15DeviceState]):
+    """Class to coordinate multiple CCM15Climate devices."""
+
+    def __init__(self, hass: HomeAssistant, host: str, port: int) -> None:
+        """Initialize the coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=host,
+            update_method=self._async_update_data,
+            update_interval=datetime.timedelta(seconds=DEFAULT_INTERVAL),
+        )
+        self._ccm15 = CCM15Device(host, port, DEFAULT_TIMEOUT)
+        self._host = host
+
+    def get_host(self) -> str:
+        """Get the host."""
+        return self._host
+
+    async def _async_update_data(self) -> CCM15DeviceState:
+        """Fetch data from Rain Bird device."""
+        try:
+            return await self._fetch_data()
+        except httpx.RequestError as err:  # pragma: no cover
```

Comment:
I think the coordinator already catches this. I know it catches aiohttp exceptions, not sure on httpx

Reply:
It does not, can you double check on your end?

Reply:
Apparently it does not